source/dub/recipe/sdl.d

```diff
@@ -174,6 +197,75 @@ private void parseConfiguration(Tag t, ref ConfigurationInfo ret, string package
+	ret.add(config.buildSettings.toSDL());
+	return ret;
+}
+
+private Tag[] toSDL(in ref BuildSettingsTemplate bs)
+{
+	Tag[] ret;
+	void add(string name, string value) { ret ~= new Tag(null, name, [Value(value)]); }
+	void adda(string name, string suffix, in string[] values) {
+		ret ~= new Tag(null, name, values[].map!(v => Value(v)).array,
+			suffix.length ? [new Attribute(null, "platform", Value(suffix[1 .. $]))] : null);
+	}
+
+	string[] toNameArray(T, U)(U bits) if(is(T == enum)) {
+		string[] ret;
+		foreach (m; __traits(allMembers, T))
+			if (bits & __traits(getMember, T, m))
+				ret ~= m;
+		return ret;
+	}
+	
+	foreach (pack, d; bs.dependencies) {
+		Attribute[] attribs;
+		if (d.path.length) attribs ~= new Attribute(null, "path", Value(d.path.toString()));
+		else attribs ~= new Attribute(null, "version", Value(d.versionString));
+		if (d.optional) attribs ~= new Attribute(null, "optional", Value(true));
+		ret ~= new Tag(null, "dependency", [Value(pack)], attribs);
+	}
+	if (bs.systemDependencies !is null) add("systemDependencies", bs.systemDependencies);
+	if (bs.targetType != TargetType.autodetect) add("targetType", bs.targetType.to!string());
+	if (bs.targetPath.length) add("targetPath", bs.targetPath);
+	if (bs.targetName.length) add("targetName", bs.targetName);
+	if (bs.workingDirectory.length) add("workingDirectory", bs.workingDirectory);
+	if (bs.mainSourceFile.length) add("mainSourceFile", bs.mainSourceFile);
+	foreach (pack, conf; bs.subConfigurations) ret ~= new Tag(null, "subConfiguration", [Value(pack), Value(conf)]);
+	foreach (suffix, arr; bs.dflags) adda("dflags", suffix, arr);
+	foreach (suffix, arr; bs.lflags) adda("lflags", suffix, arr);
+	foreach (suffix, arr; bs.libs) adda("libs", suffix, arr);
+	foreach (suffix, arr; bs.sourceFiles) adda("sourceFiles", suffix, arr);
+	foreach (suffix, arr; bs.sourcePaths) adda("sourcePaths", suffix, arr);
+	foreach (suffix, arr; bs.excludedSourceFiles) adda("excludedSourceFiles", suffix, arr);
+	foreach (suffix, arr; bs.copyFiles) adda("copyFiles", suffix, arr);
+	foreach (suffix, arr; bs.versions) adda("versions", suffix, arr);
+	foreach (suffix, arr; bs.debugVersions) adda("debugVersions", suffix, arr);
+	foreach (suffix, arr; bs.importPaths) adda("importPaths", suffix, arr);
+	foreach (suffix, arr; bs.stringImportPaths) adda("stringImportPaths", suffix, arr);
+	foreach (suffix, arr; bs.preGenerateCommands) adda("preGenerateCommands", suffix, arr);
+	foreach (suffix, arr; bs.postGenerateCommands) adda("postGenerateCommands", suffix, arr);
+	foreach (suffix, arr; bs.preBuildCommands) adda("preBuildCommands", suffix, arr);
+	foreach (suffix, arr; bs.postBuildCommands) adda("postBuildCommands", suffix, arr);
+	foreach (suffix, bits; bs.buildRequirements) adda("buildRequirements", suffix, toNameArray!BuildRequirement(bits));
+	foreach (suffix, bits; bs.buildOptions) adda("buildOptions", suffix, toNameArray!BuildOption(bits));
```

Comment:
I'd like to postpone that to a separate PR - and then also for de-serialization and for JSON.